src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -86,7 +85,9 @@ def start_new(self):
         except queue.Empty:
             logger.info(f"Process {process.pid} did not send ready message, exit.")
             self.end()
-            self.start_new()
+            # If there are no more tasks, the process is not re-created
+            if not task_queue.empty():
```

Comment:
What would happen if we start new process when task queue is empty?

Reply:
It includes two cases:
1、After a limited number of restarts, the new process is ready, and at this time, if task queue is empty, the new process will be killed at once in _timeout_process_wrapper fun, which means the new process was killed without doing anything.
2、The process was never ready, it kept restarting, which takes up some system resources.

So, restarting the process only when task queue is not empty which will avoid the above two case.